### PR TITLE
perf: Speed-up last_committed_at computation in metrics_fileinfo

### PR DESCRIPTION
**Pull request by radomirbosak**
### Summary or Background

This PR increases the performance of the `last_committed_at` value computation in `metrics_fileinfo` runner.

Instead of calling `git log` for each file separately, we dump the entire git history and parse the last commit date from there.

I tested the new algorithm performance on localhost on the [torvalds/linux](https://github.com/torvalds/linux) repository:

| torvalds/linux (~74000 files) | `last_committed_at` computation duration |
| --- | --- |
| before |  336 seconds | 
| after | 72 seconds (79% time reduction) | 

### Changes overview

1. The main change happened in `lib/runners/processor/metrics_fileinfo.rb`.
Other less significant changes are in the
2. `metrics_fileinfo.rbs` signature file 
3.  and one new smoke test was added. (`tests/smokes/*`)

### CODE REVIEW

**Comment by radomirbosak** [comment]
I tested this successfully on localhost. Next I'll need to deal with steep issues.

**Comment by gnakagaw** [comment]
Sorry. I've closed this PR by mistake. Reopened.

File: lib/runners/processor/metrics_fileinfo.rb [hunk 1]
```diff
@@ -111,11 +111,58 @@ def last_committed_at
...
+      date, files_str = commit_line.split("\n", 2)
+      raise "Commit date could not be determined." if date.nil?
+
+      date = date.split("\0")[-1]  # commits with no filechanges stick to the next commit in output, pick the right one
+      files_str = files_str || ""  # in case commit has no changed files, files_str could be nil
+      [date, files_str.split("\0")]
+    end
+
+    def datemax(first, second)
+      # Return the greater (later) of two iso8601 date strings, with empty string always counting as earlier one
```
Comment (radomirbosak) [inline lib/runners/processor/metrics_fileinfo.rb:136]: > __[Goodcheck]__ Consider adding `fraction_digits` to when using `Time#iso8601`. ([view](https://sider.review/gh/repos/202486306/pulls/2866#issue-6406992))
>
> <details>
>
> In most cases, milliseconds are necessary. For example:
>
> ```ruby
> Time.now.iso8601    #=> "2021-01-08T11:35:05+09:00"
> Time.now.iso8601(3) #=> "2021-01-08T11:35:05.371+09:00"
> ```
>
> See also <https://ruby-doc.org/stdlib-2.7.2/libdoc/time/rdoc/Time.html#method-i-iso8601>.
>
> | Rule |
> | ---- |
> | `sider.runners.time_iso8601` |
>
> You can [close this issue](https://sider.review/gh/repos/202486306/pulls/2866#issue-6406992) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

This is a false positive. I am not using date formatting, but parsing instead.

**Comment by radomirbosak** [comment]
I've resolved steep issues added a smoke test and re-rested this on localhost.

**Comment by radomirbosak** [comment]
This is now ready for review.

**Review by gnakagaw — APPROVED** [review]

**Pull request merged**